### PR TITLE
Payload-Frontend

### PR DESCRIPTION
**Pull request by SumairaRasheed**
Header Responsive Screen

### CODE REVIEW

**Comment by haris-01** [comment]
can we change the target to develop instead of master?

File: src/blocks/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <div>
+      <div className="flex bg-white">
+        <div className="relative overflow-hidden max-w-xl pl-20 pr-20 pb-10">
+          <div>
+            <h2 className="font-bold text-3xl mb-4 text-blue-500">{heading}</h2>
+            <p className="text-lg text-black">{text}</p>
+          </div>
+        </div>
+      </div>
+    </div>
```
Comment (haris-01) [inline src/blocks/Home.jsx:16]: seems like this component has unnecessary div elements.

File: src/blocks/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import Image from "next/image";
+import TwoColumn from "./TwoColumn";
```
Comment (haris-01) [inline src/blocks/Home.jsx:3]: not using this is this file. can we remove it?

File: scripts/fetchGlobalData.js [hunk 3]
```diff
@@ -1,53 +1,51 @@
...
+loadEnvConfig(process.cwd());
 
+async function fetchGlobalData() {
+  try {
+    console.log("Running Initial Fetch");
+    const [header, footer] = await Promise.all([
+      axios(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/globals/header`).then(
+        (res) => res?.data
+      ),
+      axios(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/globals/footer`).then(
```
Comment (haris-01) [inline scripts/fetchGlobalData.js:15]: this file seems to be a global function to fetch data. but all it's doing is fetching header and footer content. can we make this dynamic?

File: scripts/fetchGlobalData.js [hunk 4]
```diff
@@ -1,53 +1,51 @@
...
+const { loadEnvConfig } = require("@next/env");
 
-loadEnvConfig(process.cwd())
+loadEnvConfig(process.cwd());
 
+async function fetchGlobalData() {
+  try {
+    console.log("Running Initial Fetch");
+    const [header, footer] = await Promise.all([
+      axios(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/globals/header`).then(
```
Comment (haris-01) [inline scripts/fetchGlobalData.js:12]: which should not use the combination of .then() and await. we should prefer asych await approach.

File: src/blocks/SimpleRichText.jsx [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import Image from "next/image";
+import React from "react";
+
+export default function SimpleRichText({ body }) {
+  return <div>{body}</div>;
```
Comment (haris-01) [inline src/blocks/SimpleRichText.jsx:5]: what is the purpose of this? doesn't seem to do anything.

File: src/blocks/global/Header.jsx [hunk 6]
```diff
@@ -1,33 +1,33 @@
-import React from 'react'
-import { useState } from 'react'
-import header from '@/globalData/header.json'
-import Image from 'next/image'
-import Link from 'next/link'
+import React from "react";
+import header from "@/globalData/header.json";
```
Comment (haris-01) [inline src/blocks/global/Header.jsx:2]: af we fetching the content or importing the content? not sure about the approach here.

File: src/blocks/global/Header.jsx [hunk 7]
```diff
@@ -1,33 +1,33 @@
...
+import Image from "next/image";
+import Link from "next/link";
+import "tailwindcss/tailwind.css";
```
Comment (haris-01) [inline src/blocks/global/Header.jsx:5]: this seems irrelevant event here. we should have tailwind setup globally.

File: src/blocks/global/Header.jsx [hunk 8]
```diff
@@ -1,33 +1,33 @@
...
+        }
+        transformedData.navLinks[parentIndex].childrens.push(transformedItem);
       } else {
-          // No parent, add directly to the main array
-          transformedData.navLinks.push(transformedItem);
+        transformedData.navLinks.push(transformedItem);
       }
+    } else {
+      transformedData.navLinks.push(transformedItem);
+    }
```
Comment (haris-01) [inline src/blocks/global/Header.jsx:30]: I think all of this transformation can be done by native javascript methods like filter or reduce. which would make it a lok simpler and cleaner.

File: src/blocks/global/Header.jsx [hunk 9]
```diff
@@ -36,44 +36,65 @@ function transformData(data) {
...
+              alt={header.logo.alt}
+            />
+          </Link>
         </div>
-        <ul className="flex gap-10 p-6 absolute md:relative flex-col md:flex-row top-[-100%] md:top-[inherit]">
-            {newData.navLinks.map((link) => {  
-        if (link.childrens) {
-            // Render submenu
-            return (
+        <ul className="hidden lg:flex gap-12 text-black">
```
Comment (haris-01) [inline src/blocks/global/Header.jsx:63]: IMO the UI logic can be put in a reusable code to ensure consistency. lists styles usually don't change throughout  the app

File: src/blocks/global/NavMenu.jsx [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        transformedData.navLinks[parentIndex].childrens.push(transformedItem);
+      } else {
+        transformedData.navLinks.push(transformedItem);
+      }
+    } else {
+      transformedData.navLinks.push(transformedItem);
+    }
+  });
+
+  return transformedData;
```
Comment (haris-01) [inline src/blocks/global/NavMenu.jsx:39]: same as mentioned before to use native methods.

**Comment by haris-01** [comment]
the branch name doesn't correspond\ with the content in it. I suggest keeping the changes specific to branch name to avoid confusion

File: src/blocks/Home.jsx (hunk 2 above)
Comment (SumairaRasheed) [reply src/blocks/Home.jsx:3]: we are using this file on all pages .

File: src/blocks/global/Header.jsx (hunk 6 above)
Comment (SumairaRasheed) [reply src/blocks/global/Header.jsx:2]: Here we are importing header.json file which fetch data globally from the payload and the data we are using in Header.jsx file .

File: scripts/fetchGlobalData.js (hunk 3 above)
Comment (SumairaRasheed) [reply scripts/fetchGlobalData.js:15]: This file was already made dynamic, yes, but its code can be further optimized, which I have done.

File: scripts/fetchGlobalData.js (hunk 4 above)
Comment (SumairaRasheed) [reply scripts/fetchGlobalData.js:12]: Changed according to the async await approach.

File: src/blocks/SimpleRichText.jsx (hunk 5 above)
Comment (SumairaRasheed) [reply src/blocks/SimpleRichText.jsx:5]: this is just for time being.

File: src/blocks/global/Header.jsx (hunk 7 above)
Comment (SumairaRasheed) [reply src/blocks/global/Header.jsx:5]: resolved

File: src/blocks/global/Header.jsx (hunk 8 above)
Comment (SumairaRasheed) [reply src/blocks/global/Header.jsx:30]: The transformData function uses the reduce method to iterate over data.navLinks and build the transformed structure more concisely to make the code cleaner and more readable.

File: src/blocks/global/NavMenu.jsx (hunk 10 above)
Comment (SumairaRasheed) [reply src/blocks/global/NavMenu.jsx:39]: Resolved.
